Yank/multistate/sams.py

```diff
@@ -559,7 +559,7 @@ def _update_stage(self):
             else:
                 raise ValueError("Unknown flatness_criteria %s" % flatness_criteria)
 
-            if advance:
+            if advance or ((self._t0 > 0) and (self._iteration > self._t0)):
```

Comment:
Just to understand this better, in what situation we can have `self._t0 > 0` and `self._iteration <= self._t0` at the same time?

Reply:
Good point, but once we start adding the ability to back up by the checkpoint interval to recover, I'm not sure if just checking `self._t0 > 0` is sufficient anymore

Reply:
Ah, I see, good point.